### PR TITLE
Build: adds signing config to gradle 

### LINKED ISSUE #1123: Build: add Gradle task+plugin for signing
# Feature Request

Add the `signing` gradle plugin to the root buildfile and every module:
```kotlin
plugins {
    signing
}
```

Add a `signing` configuration to every module that produces an artifact:
```kotlin
publication { 
    // should already be there
}
// add this:
signing {
    sign(publishing.publications["publicationName"]) 
}
```

check documentation [here](https://docs.gradle.org/current/userguide/signing_plugin.html#sec:signatory_credentials)

## Which Areas Would Be Affected?
Build

## Why Is the Feature Desired?
MavenCentral requires artifacts be signed with a GPG key, which is done on a Jenkins instance hosted by the Eclipse Foundation.

## Type of Issue
New Feature

## Checklist

- [x] assigned appropriate label?
- [ ] **Do NOT select a milestone or an assignee!**

### LINKED ISSUE #1147: Chore: add signing configuration to all artifacts
# Feature Request

The following information must be added to the build file(s):
- POM information:
  - project name
  - project description
  - scm connection
  - scm url
  - developer id, name, email
  - license name+url
- general signing configuration
- Maven repo for OSSRH
- `withJavadocJar()`
- `withSourcesJar()`

in addition, every artifact has to declare what should be signed, for example:
```kotlin
publishing {
    publications {
        create<MavenPublication>("core-base") {
            artifactId = "core-base"
            from(components["java"])
        }
    }
    // add this:
    signing {
        sign("core-base")
    }
}
```

_this will touch EVERY build file that produces an artifact!_

## Which Areas Would Be Affected?
build files

## Why Is the Feature Desired?
required for MavenCentral builds

## Type of Issue
chore

## Checklist

- [x] assigned appropriate label?
- [\ ] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

Adds publication config, such as project name, description, etc. plus the OSSHR Maven repository as well as signing configs to the build file.

## Why it does that

Required by OSSRH/MavenCentral

## Todo
- Should the signing stuff get moved into its own plugin?

## Further notes:
- `./gradlew publishToMavenLocal` now takes very long (>5min on an Intel i9), and it seems not to be multi-thread-able. It's probably the signing that actually takes so long.

## Linked Issue(s)

Closes #1147 
Closes #1123 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -34,3 +33,9 @@ httpMockServer=5.12.0
 micrometerVersion=1.8.2
 bouncycastleVersion=1.70
 gatlingVersion=3.7.5
+edcDeveloperId=mspiekermann
+edcDeveloperName=Markus Spiekermann
+edcDeveloperEmail=markus.spiekermann@isst.fraunhofer.de
```
Comment (paullatzelsperger) [inline gradle.properties:38]: @mspiekermann are you cool with adding you as "the man"?

**Review by chrislomonico — COMMENTED** [review]
Reviewed changes

- files needing correction for doc generation have been addressed and match up with the records I have
- pulling branch and confirming that the execution runs as expected generating necessary artifacts

**Comment by paullatzelsperger** [comment]
@jimmarino @juliapampus @MoritzKeppler @mspiekermann how do we proceed with this? should we merge it and try to get it to build on JIPP?

File: gradle.properties (hunk 1 above)
Comment (chrislomonico) [reply gradle.properties:38]: It is possible to provide more than one developer

**Review by chrislomonico — APPROVED** [review]
Completed confirmation that all necessary artifacts are being signed with local build and against a  Jenkins local instance.  Per previous comments the rest of the files that needed to be modified (contained formatting errors) align with the list of those I found.

**Pull request merged**